controllers/membersController.js

```diff
@@ -9,36 +10,87 @@ const db = require('../utils/firestore')
  */
 
 async function getMembers (req, res) {
-  const snapshot = await db.collection('members').get()
+  try {
+    const snapshot = await db.collection('members').get()
 
-  const allMembers = []
+    const allMembers = []
 
-  snapshot.forEach((doc) => {
-    allMembers.push({
-      id: doc.id,
-      ...doc.data()
+    snapshot.forEach((doc) => {
+      allMembers.push({
+        id: doc.id,
+        ...doc.data()
+      })
     })
-  })
 
-  res.json(allMembers)
+    if (allMembers.length) {
+      res.json({
+        message: 'Members returned successfully!',
+        members: allMembers
+      })
+    }
+
+    res.boom.notFound('No members available')
+  } catch (error) {
+    logger.error(`Error while fetching all members: ${error}`)
+    res.boom.serverUnavailable('Something went wrong please contact admin')
+  }
 }
 
-// #TODO
-async function addNewMember (memberData) {
-  const memberId = 'test'
-  const memberRef = db.collection('members').doc(memberId)
-  const doc = await memberRef.get()
-
-  if (!doc.exists) {
-    await memberRef.set({
-      first: 'First Name',
-      last: 'Last Name',
-      born: Math.random()
-    })
+/**
+ * Fetches the data about member with given id
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ * @param next {Function} - Express middleware function
+ */
+
+async function getMember (req, res) {
+  try {
+    const doc = await db.collection('members').doc(req.params.id).get()
+
+    if (doc.exists) {
+      res.json({
+        message: 'Member returned successfully!',
+        member: doc.data()
+      })
+    }
+
+    res.boom.notFound('Member doesn\'t exist')
+  } catch (error) {
+    logger.error(`Error while fetching all members: ${error}`)
+    res.boom.serverUnavailable('Something went wrong please contact admin')
+  }
+}
+
+/**
+ * Add new member
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - Member object
+ * @param res {Object} - Express response object
+ * @param next {Function} - Express middleware function
+ */
+async function addNewMember (req, res) {
+  try {
+    const memberRef = await db.collection('members').doc(req.body.id)
```

Comment:
How will we be creating a relationship between `users` collection and `members`? The data that this API is storing, part of it like github_id, github_display_name is already added in users collection.

Reply:
the is_member boolean will take care of this part